docs/cloud/satellites/self-hosted.md

```diff
@@ -0,0 +1,215 @@
+* `EARTHLY_ORG` - the name of your Earthly Organization created on [https://cloud.earthly.dev](https://cloud.earthly.dev)
+* `SATELLITE_NAME` - a name chosen by the user to identify the satellite instance
+* `SATELLITE_HOST` - Hostname or IP address for the Earthly CLI to connect. Note that users will select the satellite by name when running builds.
+
+Using a volume for cache storage is recommended for preserving cache after the container is destroyed. See [Advanced Configuration](#advanced-configuration) below for more details and a list of optional environment variables that can be used to finetune your deployment. See also [Platform Specific Guides](#platform-specific-guides) for examples of deploying to platforms like Kubernetes or EC2.
+
+The self-hosted satellite will not accept localhost as an address. For instructions on testing locally, please refer to [Running on Localhost](#running-on-localhost) below.
+
+
+{% hint style='info' %}
+##### Security Advisory
+The `--privileged` flag is currently required. Privileged mode is used by some features of earthly, including the `WITH DOCKER` command in Earthfiles. If using privileged mode is a concern, running satellites in a dedicated VM or Kubernetes cluster can provide better isolation compared to running the container directly in your production environment.
+{% endhint %}
+
+### Auto-discovery of Host _\*experimental\*_
+
+When `SATELLITE_HOST` is left unset, the satellite will attempt to auto-discover its host address. This feature may not always work, and currently only supports environments using cloud-init (such as EC2).
+
+## Connecting to a Self-Hosted Satellite
+
+Once the satellite has finished registering, it can be selected and managed using typical earthly satellite commands from the CLI. 
+For example, to select the satellite for use in subsequent builds:
+
+```
+earthly sat select my-satellite
+```
+
+Or, to invoke a build once on the satellite, without persistently selecting it:
+
+```
+earthly --sat my-satellite +my-target
+```
+
+Note that `earthly-v0.8.0` or later is required to connect to a self-hosted satellite.
+
+## Managing Self-Hosted Satellites
+
+A list of satellites in your org be viewed via the satellite ls command. For example:
+
+```
+> earthly sat ls
+NAME          PLATFORM     SIZE         VERSION  STATE
+my-satellite  linux/amd64  self-hosted  v0.8.3   operational
+...
+```
+
+The size field of a self-hosted satellite will be listed as `self-hosted`, as opposed to one of the sizes of Earthly Cloud’s managed satellites.
+
+The state field will be either `operational` or `offline`. A self-hosted satellite automatically transitions to an `offline` state when it is gracefully terminated, or if no heartbeat has been received by Earthly’s servers for a while.
+
+You can run `satellite rm` on a self-hosted satellite when it is in an offline state to remove it from your account permenantly.
```

Comment:
Add mention of `--force`?

Reply:
Not sure if we should promote that for self-hosted. I think of it as an escape hatch just in case something gets stuck in a bad state, as opposed to normal operation.

Reply:
It's probably a very useful escape hatch - sometimes you don't have control over your infrastructure and you really need to disconnect it. Anyway... it's a small feature, we can also just leave it out and let them find it via `rm` attempt - presumably the error says that you can use `--force`.

Reply:
Yes, it will say that they can re-run with force if they try it. 

Also (in case it wasn't obvious) this wouldn't trigger the actual self-hosted satellite to terminate. Just remove it from the list on earthly side.